lib/net_buf/buf.c

```diff
@@ -121,7 +141,7 @@ static uint8_t *mem_pool_data_alloc(struct net_buf *buf, size_t *size,
 	*ref_count = 1U;
 
 	/* Return pointer to the byte following the ref count */
-	return ref_count + sizeof(void *);
+	return ref_count + GET_ALIGN(buf_pool);
```

Comment:
This is a waste of RAM on a platform where it must be aligned to the DCACHE line size. I prefer not to store any metadata in the buffer space.